src/main/java/org/gitlab4j/api/GitLabApiClient.java

```diff
@@ -470,6 +470,35 @@ protected Response head(MultivaluedMap<String, String> queryParams, URL url) {
         return (invocation(url, queryParams).head());
     }
 
+    /**
+     * Perform an HTTP PATCH call with the specified query parameters and path objects, returning
+     * a ClientResponse instance with the data returned from the endpoint.
+     *
+     * @param queryParams multivalue map of request parameters
+     * @param pathArgs variable list of arguments used to build the URI
+     * @return a ClientResponse instance with the data returned from the endpoint
+     * @throws IOException if an error occurs while constructing the URL
+     */
+    protected Response patch(MultivaluedMap<String, String> queryParams, Object... pathArgs) throws IOException {
+        URL url = getApiUrl(pathArgs);
+        return (patch(queryParams, url));
+    }
+
+    /**
+     * Perform an HTTP PATCH call with the specified query parameters and URL, returning
+     * a ClientResponse instance with the data returned from the endpoint.
+     *
+     * @param queryParams multivalue map of request parameters
+     * @param url the fully formed path to the GitLab API endpoint
+     * @return a ClientResponse instance with the data returned from the endpoint
+     */
+    protected Response patch(MultivaluedMap<String, String> queryParams, URL url) {
+        Entity<?> empty = Entity.text("");
+        // use "X-HTTP-Method-Override" header on POST to override to unsupported PATCH
+        return (invocation(url, queryParams)
```

Comment:
Hello thanks for your PR. 

I did not check if it works, but I think you can use `invocation(url, queryParams).method("PATCH", empty);`

Can you try with my proposal ?

Reply:
Thanks for the suggestion, @jabby. In fact I tried that first (after not finding a native patch() method) and found it doesn't work with the current client, and instead returns the following exception:
```
org.gitlab4j.api.GitLabApiException: java.net.ProtocolException: Invalid HTTP method: PATCH
```
There is some [discussion](https://stackoverflow.com/questions/22355235/patch-request-using-jersey-client) of other workarounds for the [underlying bug](https://bugs.openjdk.org/browse/JDK-8207840), but this approach seemed sufficient and less problematic to me. I'm open to pushback on that.

Also, to help in the future and at least exercise the method, I've added a test handling the current tested version with comments on what should happen with Premium.

Reply:
Any other questions on this, @jabby ?